toodlext/src/main.rs

```diff
@@ -0,0 +1,355 @@
+        let labels = match self.labels {
+            Some(ref labels) => labels.iter().map(|label| label.to_label()).collect(),
+            None => Vec::new(),
+        };
+        Item {
+            id: None,
+            uuid: self.uuid
+                .clone()
+                .and_then(|uuid| Uuid::parse_str(&uuid).ok())
+                .unwrap_or_else(|| Uuid::nil()),
+            name: self.name.clone(),
+            due_date,
+            completion_date,
+            labels,
+        }
+    }
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(tag = "type")]
+enum RequestBody {
+    CreateTodo(ItemInfo),
+    GetTodos,
+    TodoChangeName { uuid: String, name: String },
+    TodoChangeDueDate {
+        uuid: String,
+
+        #[serde(rename = "dueDate")]
+        due_date: i64,
+    },
+    TodoChangeCompletionDate {
+        uuid: String,
+
+        #[serde(rename = "completionDate")]
+        completion_date: i64,
+    },
+    TodoAddLabel { uuid: String, label: LabelInfo },
+    TodoRemoveLabel { uuid: String, name: String },
+    RemoveTodo { uuid: String },
+    AddLabel(LabelInfo),
+    RemoveLabel { name: String },
+    GetLabels,
+}
+
+#[derive(Deserialize, Debug)]
+struct Request {
+    id: i64,
+    body: RequestBody,
+}
+
+impl Request {
+    fn read_from(input: &mut StdinLock) -> Result<Request, Error> {
+        match input.read_u32::<NativeEndian>() {
+            Ok(length) => {
+                let mut message = input.take(length as u64);
+                let mut buffer = Vec::with_capacity(length as usize);
+                eprintln!("Reading request from browser");
+                match message.read_to_end(&mut buffer) {
+                    Ok(_) => {
+                        serde_json::from_slice(&buffer).map_err(|err| {
+                            eprintln!("Error parsing request payload {:?}: {:?}",
+                                      String::from_utf8_lossy(&buffer),
+                                      err);
+                            Error::BadJSON
+                        })
+                    }
+                    _ => Err(Error::IOError),
+                }
+            }
+            _ => Err(Error::IOError),
+        }
+    }
+}
+
+#[derive(Serialize, Debug)]
+struct Response {
+    id: i64,
+    ok: bool,
+    body: ResponseBody,
+}
+
+impl Response {
+    fn write_to(&self, output: &mut StdoutLock) -> io::Result<()> {
+        let message = serde_json::to_vec(self)?;
+        output.write_u32::<NativeEndian>(message.len() as u32)?;
+        output.write_all(&message)?;
+        output.flush()
+    }
+}
+
+#[derive(Serialize, Debug)]
+#[serde(untagged)]
+enum ResponseBody {
+    CreateTodo(ItemInfo),
+    UpdateTodo(ItemInfo),
+    GetTodos(Vec<ItemInfo>),
+    GetLabels(Vec<LabelInfo>),
+    RemoveTodo { uuid: String },
+    AddLabel(LabelInfo),
+    RemoveLabel { name: String },
+    Err(Error),
+}
+
+fn update_item_by_uuid(toodle: &mut Toodle,
```

Comment:
`master` branch now has `toodle_update_item_by_uuid`, which should work well here.

Reply:
`toodle_update_item_by_uuid` is the unsafe FFI function, right? I'll add `update_item_by_uuid` to `Toodle`, and have the bridge and `toodle_update_item_by_uuid` use it.